Add gateway and sui binary support to split a coin to N coins

aschran opened the pull request:
Fixes #1220.

Manually tested with local sui binary using command:
sui client split-coin --count 3 <...>

Clay-Mysten reviewed (approved):
Great work on cli-client.md, Andrew! That Markdown change looks good to me with some small changes I added, including a summary of results I added reading, "From the result, we can see three coins with values of roughly one-third of 100000."

Please take a look at my commit and improve as you desire. Thanks!

oxade commented on crates/sui-cost/src/empirical_transaction_cost.rs:
```diff
@@ -31,6 +31,7 @@ pub enum CommonTransactionCosts {
     Publish,
     MergeCoin,
     SplitCoin(usize),
+    SplitCoinEqual,
```
You should extend this to take the number of coins, and provide a few runs like the SplitCoin case.
This helps us track multiple data points

oxade replied:
Thanks @aschran. For more context on why we want multiple points:
- if we find that the cost grows fairly linearly with the number of splits, we can build formula-based cost estimator which allows apps and clients to quickly estimate what their splits might cost at any arbitrary N.

aschran commented:
> Great work on cli-client.md, Andrew! That Markdown change looks good to me with some small changes I added, including a summary of results I added reading, "From the result, we can see three coins with values of roughly one-third of 100000."
> 
> Please take a look at my commit and improve as you desire. Thanks!

LGTM!

oxade commented:
@aschran not a blocker but do you think we stop folks early in the sdk etc from passing in 0 as number of splits?
